### PR TITLE
chore: only ignore top level directories in gitignore for some cases

### PR DESCRIPTION
**Pull request by jviotti**
We accidentally added build/ to gitignore, causing any new files from
`scripts/build` to be ignored.

See: https://github.com/resin-io/etcher/pull/1547#discussion_r126790010
Signed-off-by: Juan Cruz Viotti <jv@jviotti.com>

### CODE REVIEW

File: scripts/build/target-convert.sh [hunk 1]
```diff
@@ -0,0 +1,67 @@
+#!/bin/bash
+
+###
+# Copyright 2016 resin.io
```
Comment (lurch) [inline scripts/build/target-convert.sh]: Should be 2017 ;)

**Comment by lurch** [comment]
Doh! Looks like this bug has been there since 2nd May https://github.com/resin-io/etcher/commit/b75dfd3ecedf5f1218d00c8988085e562f2bce19 :(

Are there any other file-paths in `.gitignore` that _ought_ to use an 'absolute' path?

File: scripts/build/target-convert.sh (hunk 1 above)
Comment (jviotti) [inline scripts/build/target-convert.sh]: Done!

**Comment by jviotti** [comment]
Ugh, that sucks. I'll revise `gitignore` and update this PR accordingly.

File: .gitignore [hunk 2]
```diff
@@ -21,15 +21,15 @@ coverage
 .lock-wscript
 
 # Compiled binary addons (http://nodejs.org/api/addons.html)
-build
+/build
 
 # Dependency directory
 # https://docs.npmjs.com/misc/faq#should-i-check-my-node-modules-folder-into-git
 node_modules
 bower_components
```
Comment (lurch) [inline .gitignore]: Do we still need this line, if we're not using bower any more?

File: .gitignore [hunk 3]
```diff
@@ -21,15 +21,15 @@ coverage
 .lock-wscript
 
 # Compiled binary addons (http://nodejs.org/api/addons.html)
-build
+/build
 
 # Dependency directory
 # https://docs.npmjs.com/misc/faq#should-i-check-my-node-modules-folder-into-git
 node_modules
```
Comment (lurch) [inline .gitignore]: Should this have a leading slash too?

File: .gitignore [hunk 4]
```diff
@@ -21,15 +21,15 @@ coverage
...
 bower_components
 
 # Compiled Electron releases
```
Comment (lurch) [inline .gitignore]: Electron -> Etcher ?

File: .gitignore (hunk 3 above)
Comment (jviotti) [inline .gitignore]: I thought about that, but I don't think a directory name such as `node_modules` could ever be the source of confusion, plus this rule would cover us from accidental nested `node_modules` in other directories.

File: .gitignore (hunk 2 above)
Comment (jviotti) [inline .gitignore]: Good catch. I'll get rid of it!

File: .gitignore (hunk 4 above)
Comment (jviotti) [inline .gitignore]: Fixed!

**Pull request merged**